### ISSUE TITLE
Ensure that the error event is the last event to be fired

### ISSUE THREAD

**Issue by foolip** [w3c/mediacapture-record#52]
http://w3c.github.io/mediacapture-record/MediaRecorder.html#general-principles

> If recording has been started and not yet stopped when the error occurs, then after raising the error, the UA must fire a `dataavailable event`, containing any data that it has gathered, and then a stop event.

This seems odd, it will mean that sometimes there will be data after the error event and sometimes there will not. How about instead requiring that if there is any data, that the `dataavailable` event be fired before the `error` event? Assuming that both the data and the error originates from the same bits of code then that ought to also match the actual order things happen in.

**Comment by yell0wd0g** [comment]
I agree, but this seems to be implemented by both Gecko ([here](https://github.com/mozilla/gecko-dev/blob/master/dom/media/MediaRecorder.cpp#L181) and [here](https://github.com/mozilla/gecko-dev/blob/master/dom/media/MediaRecorder.cpp#L210) -- note that the error is not followed by an `return NS_ERROR_FAILURE`) and Chromium ([here](https://cs.chromium.org/chromium/src/third_party/WebKit/Source/modules/mediarecorder/MediaRecorder.cpp?q=mediarecorder.cpp&sq=package:chromium&dr&l=336)).

The Section is now non-normative. What do you think we should do?

**Comment by foolip** [comment]
Just changing the order of the events there should fix the problem, implementation-wise I guess it's simple enough?

**Comment by yell0wd0g** [comment]
@foolip, I think I should homogenise all these sequences so that the Error event is the last one; potentially after an ondataavailable event and/or an onstop event, sgty?

**Comment by foolip** [comment]
Yep, firing the error event last is exactly what I'm asking for :)
